### PR TITLE
feat(dispose): Added Disposable interface

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
- The "what"; a concise description of each logical change
- Another change

The "why", or other context.

## Issues
- "THING-1234" or "Fixes #123"

### CODE REVIEW

File: OptimizelySDK/Optimizely.cs [hunk 1]
```diff
@@ -781,5 +782,14 @@ private object GetTypeCastedVariableValue(string value, FeatureVariable.Variable
...
         }
-    }
+
+        public void Dispose()
+        {
+            if (Disposed) return;
+
+            Disposed = true;
+            (ProjectConfigManager as IDisposable)?.Dispose();
+            ProjectConfigManager = null;
```
Comment (mikeng13) [inline OptimizelySDK/Optimizely.cs]: If we dispose the optimizely client, but then accidentally end up using it again afterwards, will this result in a null pointer exception?

Comment (msohailhussain) [reply OptimizelySDK/Optimizely.cs]: Added test case here. 
https://github.com/optimizely/csharp-sdk/pull/170/files#diff-8660e0a3c8615b111f08b9e4ab4f9b36R3485

Comment (mikeng13) [reply OptimizelySDK/Optimizely.cs]: 👍

**Comment by mikeng13** [comment]
@msohailhussain the unit test failure seems legit

**Pull request merged**